### PR TITLE
tidy rlm_mschap mod_authenticate

### PR DESCRIPTION
**Pull request by mcnewton**
mod_authenticate is soooo long :( and I can't contemplate doing anything else with it until it's been tidied up somewhat. Move some chunks of code out to separate functions.

### CODE REVIEW

File: src/modules/rlm_mschap/rlm_mschap.c [hunk 1]
```diff
@@ -1483,10 +1483,10 @@ static rlm_rcode_t mschap_error(rlm_mschap_t *inst, REQUEST *request, unsigned c
  *	find_nt_password() - try and find a correct NT-Password
  *	attribute, or calculate one if possible.
  */
-static bool CC_HINT(nonnull (1, 2, 4)) find_nt_password(rlm_mschap_t *inst,
-							REQUEST *request,
-							VALUE_PAIR *password,
-							VALUE_PAIR **ntpw)
+static inline bool CC_HINT(nonnull (1, 2, 4)) find_nt_password(rlm_mschap_t *inst,
```
Comment (alandekok) [inline src/modules/rlm_mschap/rlm_mschap.c]: inline shouldn't be used.  C compilers are generally smart enough to do the right thing.

**Comment by alandekok** [comment]
The "inline" patch should just be removed.  It isn't necesssary.  We use "inline" only rarely... if at all.

**Comment by mcnewton** [comment]
OK, thanks - done.

**Comment by arr2036** [comment]
Compiler will only inline things with optimization turned on, which means there is no inlining for developer builds unless it's explicitly declared, but that won't really make a difference.

Is there a specific argument against inlining?

File: src/modules/rlm_mschap/rlm_mschap.c [hunk 2]
```diff
@@ -1641,29 +1585,132 @@ static rlm_rcode_t CC_HINT(nonnull) mod_authenticate(void *instance, REQUEST *re
...
+ *	attributes and configuration.
+ *	We will try to find out password in configuration
+ *	or in configured passwd file.
+ *	If one is found we will check paraneters given by NAS.
+ *
+ *	If PW_SMB_ACCOUNT_CTRL is not set to ACB_PWNOTREQ we must have
+ *	one of:
+ *		PAP:      PW_USER_PASSWORD or
+ *		MS-CHAP:  PW_MSCHAP_CHALLENGE and PW_MSCHAP_RESPONSE or
+ *		MS-CHAP2: PW_MSCHAP_CHALLENGE and PW_MSCHAP2_RESPONSE
```
Comment (arr2036) [inline src/modules/rlm_mschap/rlm_mschap.c]: Could this be doxygenised?

File: src/modules/rlm_mschap/rlm_mschap.c [hunk 3]
```diff
@@ -1599,23 +1529,37 @@ static rlm_rcode_t CC_HINT(nonnull) mod_authenticate(void *instance, REQUEST *re
...
 
-	/*
-	 *	Or an LM-Password.
-	 */
+	*ntpw = nt_password;
+	return true;
+}
+
+
+/*
```
Comment (arr2036) [inline src/modules/rlm_mschap/rlm_mschap.c]: Same here

File: src/modules/rlm_mschap/rlm_mschap.c (hunk 2 above)
Comment (mcnewton) [inline src/modules/rlm_mschap/rlm_mschap.c]: Probably... that's not my comment ;) (pretty much none of it is mine - I just moved stuff around).

I'll take a look if I get a moment.

**Pull request merged**